source/reference/regular_expressions.rst

```diff
@@ -69,6 +71,9 @@ syntax described below.
 - ``{x}?``, ``{x,y}?``, ``{x,}?``: Lazy versions of the above
   expressions.
 
+- Use only standard escapes like ``\n``, ``\r``, ``\t`` and keep in mind that
+  you also need to do JSON escaping.
```

Comment:
I think this is a great addition. For the JSON string escaping part, I wonder if it would be better to put this somewhere near the top since this applies to things other than character classes as well.